src/components/navigation/header/header.tsx

```diff
@@ -1,23 +1,26 @@
 import clsx from 'clsx'
 import { Router, useRouter } from 'next/router'
-import { useState } from 'react'
+import { useEffect, useState } from 'react'
 
-import { Link } from '../content/link'
-import { MobileMenu } from '../navigation/mobile-menu'
-import { Menu } from './menu'
+import { Link } from '../../content/link'
+import { Menu } from './menu/menu'
 import { MobileMenuButton } from './mobile-menu-button'
-import { useAuthentication } from '@/auth/use-authentication'
 import { Quickbar } from '@/components/navigation/quickbar'
 import { useInstanceData } from '@/contexts/instance-context'
 
 export function Header() {
   const [mobileMenuOpen, setMobileMenuOpen] = useState(false)
-  const auth = useAuthentication()
-  const { strings, headerData } = useInstanceData()
+  const { strings } = useInstanceData()
   const router = useRouter()
 
   const hideQuickbar = router.route === '/search' || router.route === '/'
 
+  useEffect(() => {
+    document.body.addEventListener('keydown', (event) => {
+      if (event.key === 'Escape') setMobileMenuOpen(false)
+    })
+  })
```

Comment:
```suggestion
  useEffect(() => {
    document.body.addEventListener('keydown', (event) => {
      if (event.key === 'Escape') setMobileMenuOpen(false)
    })
  }, [])
```
Otherwise, you are adding a new event listener on each rerendering.

Reply:
Uh, nice catch.